addons/website_blog/views/website_blog_templates.xml

```diff
@@ -320,12 +320,12 @@ list of filtered posts (by date or tag).
                                 <t t-set="hide_date" t-value="True"/>
                             </t>
                         </span>
-                        <span t-if="len(blog_post.message_ids) > 0" class="text-nowrap ps-2 o_not_editable">|
+                        <span t-if="len(blog_post.message_ids.filtered(lambda r: r.body != '')) > 0" class="text-nowrap ps-2 o_not_editable">|
                             <i class="fa fa-comment text-muted me-1"/>
                             <a href="#discussion">
-                                <t t-esc="len(blog_post.message_ids)"/>
-                                <t t-if="len(blog_post.message_ids)>1">Comments</t>
-                                <t t-else="">Comment</t>
+                                <t t-esc="len(blog_post.message_ids.filtered(lambda r: r.body != ''))"/>
+                                <t t-if="len(blog_post.message_ids.filtered(lambda r: r.body != ''))>1"> Comments</t>
+                                <t t-else=""> Comment</t>
```

Comment:
I'm not sure how this behaves but you may need to add this as a new entry to the .pot file

Try exporting the translation template for this module and see if it introduces a space for the translation. If so *add* the entry so that both versions can be translated.